Caltech-IPAC/firefly
# FIREFLY-1359: UI Conversion: convert date/time picker to Joy UI

Fixes [FIREFLY-1359](https://jira.ipac.caltech.edu/browse/FIREFLY-1359)

- Used MUI X datepicker, removing react-datetime (note: it required adding `JoytoMUIThemeWrapper` because MUI X is made for MUI, not Joy UI yet)
- Simplified code of TimePanel (like doing form feedback rendering within it) and of DateTimeInputField (state management logic)
- Cleaned up style props no longer needed

## Testing
Both UTC and ISO mode: https://fireflydev.ipac.caltech.edu/firefly-1359-datetime-picker/firefly
- TAP Panel -> Temporal Search
- TAP service: CADC, Table: ivoa -> Timing: select overlapping specified range


Only UTC: https://firefly-1359-datetime-picker.irsakudev.ipac.caltech.edu/applications/Spitzer/SHA/
- Search by Observation Date

Check:
- making changes in datetime picker show up in datetime field
- datetime field is editable and allows copy-pasting
- making changes in datetime field show up in datetime picker (note: since minute and second use step=5, entering 02:03:04 will only show selection in hour picker - this aligns with how [MUI X datetime-picker](https://mui.com/x/react-date-pickers/date-time-picker/) works)
- entering wrong datetime in fields throws validation error like before
